pkg/trafficpolicy/trafficpolicy.go

```diff
@@ -189,18 +205,35 @@ func mergeRoutesWeightedClusters(originalRoutes, latestRoutes []*RouteWeightedCl
 	return originalRoutes
 }
 
-// subset returns true if the second array is completely contained in the first array
-func subset(first, second []string) bool {
-	set := make(map[string]bool)
-	for _, value := range first {
-		set[value] = true
+// slicesUnion returns the union of the two slices if either of the slices is a complete subset of the other
+func slicesUnion(first, second []string) []string {
```

Comment:
The `slicesUnion` naming makes it look like this function does something fairly simple.
When I read the comment, and the code, this seems really complex (and perhaps for a reason).
I wish there was a name to imply that this is some sort of a conditional union.

Reply:
I have attempted to rename the function, I hope this naming works better